Makefile

```diff
@@ -10,20 +10,17 @@ CPPRUNFLAGS:=  --enable=warning --error-exitcode=1
 CPPRUNFLAGSFILE:= --enable=performance,style --output-file=performance_and_style_report.txt
 DFF:=diff
 DFFOBJS:= test/outputs.txt test/good_outputs.txt
-JSONTST:= ./test/JsonParser_test
-OUTPTS:= ./test/generate_outputs.sh
-DCMNT:= doxygen doxconf
 alltest: runMain cppcheck cppcheckfile valgrind diff jsontst generate_outputs
 
 runMain:$(OBJS)
 	$(RUN) $(CFLAGS) -o runMain $(OBJS)
-JsonParser.o: JsonParser.cpp
+JsonParser.o: JsonParser.cpp JsonParser.h
 	$(RUN) $(CFLAGS) -c JsonParser.cpp
-Main.o: Main.cpp
+Main.o: Main.cpp 
 	$(RUN) $(CFLAGS) -c Main.cpp
-Unit.o: Unit.cpp
+Unit.o: Unit.cpp Unit.h
 	$(RUN) $(CFLAGS) -c Unit.cpp
```

Comment:
A Unit a parse miatt gondolom hasznalja a JsonParser-t. Szoval ha annak modosul a headerje, akkor is ujra kell forditani, kulonben a linker elszall.

Reply:
Szóval ha jól értettem akkor minden olyan .o fájlnál meg kell adni az összes headert amit beránt a hozzá tartozó cpp, elvileg ezt megoldottam.

Reply:
bezony, hat midnen, ami include-olva van, az kopizodik a cpp-be (errol szol az include), es ha az a header is include-ol valamit, akkor az is, es ... [az apaink apainak apait is, es az apaink apainak apainak apait is, es... ](https://videa.hu/videok/film-animacio/apaink-apainak-apait-is-i-brian-elete-J0HYjb9eaVxmrdHC)